mono/mini/aot-runtime.c

```diff
@@ -5404,58 +5404,9 @@ load_function_full (MonoAotModule *amodule, const char *name, MonoTrampInfo **ou
 					g_static_assert (MONO_TRAMPOLINE_JIT == 0);
 					target = (gpointer)mono_get_trampoline_func ((MonoTrampolineType)(ji->data.jit_icall_id - MONO_JIT_ICALL_generic_trampoline_jit));
 					break;
-#ifdef TARGET_AMD64
-				case MONO_JIT_ICALL_mono_amd64_throw_exception:
-					target = (gpointer)mono_amd64_throw_exception;
-					break;
-				case MONO_JIT_ICALL_mono_amd64_throw_corlib_exception:
-					target = (gpointer)mono_amd64_throw_corlib_exception;
-					break;
-				case MONO_JIT_ICALL_mono_amd64_resume_unwind:
-					target = (gpointer)mono_amd64_resume_unwind;
-					break;
-				case MONO_JIT_ICALL_mono_amd64_start_gsharedvt_call:
-					target = (gpointer)mono_amd64_start_gsharedvt_call;
-					break;
-#endif
-#ifdef TARGET_ARM
-				case MONO_JIT_ICALL_mono_arm_throw_exception_by_token:
-					target = (gpointer)mono_arm_throw_exception_by_token;
-					break;
-				case MONO_JIT_ICALL_mono_arm_unaligned_stack:
-					target = (gpointer)mono_arm_unaligned_stack;
-					break;
-#endif
-#if defined (TARGET_ARM) || defined (TARGET_ARM64)
-				case MONO_JIT_ICALL_mono_arm_start_gsharedvt_call:
-					target = (gpointer)mono_arm_start_gsharedvt_call;
-					break;
-				case MONO_JIT_ICALL_mono_arm_throw_exception:
-					target = (gpointer)mono_arm_throw_exception;
-					break;
-				case MONO_JIT_ICALL_mono_arm_resume_unwind:
-					target = (gpointer)mono_arm_resume_unwind;
-					break;
-#endif
-#ifdef TARGET_POWERPC
-				case MONO_JIT_ICALL_mono_ppc_throw_exception:
-					target = (gpointer)mono_ppc_throw_exception;
-					break;
-#endif
-#ifdef TARGET_X86
-				case MONO_JIT_ICALL_mono_x86_throw_exception:
-					target = (gpointer)mono_x86_throw_exception;
-					break;
-				case MONO_JIT_ICALL_mono_x86_throw_corlib_exception:
-					target = (gpointer)mono_x86_throw_corlib_exception;
-					break;
-				case MONO_JIT_ICALL_mono_x86_start_gsharedvt_call:
-					target = (gpointer)mono_x86_start_gsharedvt_call;
-					break;
-#endif
 				default:
-					g_error ("Unknown relocation '%p'\n", ji->data.target);
-					target = NULL;
+					target = mono_arch_load_function (ji->data.jit_icall_id);
+					g_assertf (target, "Unknown relocation '%p'\n", ji->data.target);
```

Comment:
Needs some kind of ifdef or needs to be added for all architectures.

Reply:
I'm "rewriting" it..

Reply:
It bugged me to lose the inlining. I think the function was provided for all architectures that support AOT, as well as PowerPC, but anyway, the rewrite is tweaked to handle architectures with no extra functions.

Perhaps I should have removed only one hash table in this PR and left the AOT one asis. I could do that, though still heading toward the same place, if ok.

Reply:
Another solution is put back the AOT registration, into the struct/array, and just index into that (`mono_find_jit_icall)`.